### DISCUSSION TITLE
Add a field for subheading in article

### DISCUSSION THREAD

**Discussion by ebbes-isch-emmer** [joomla/joomla-cms#43255] [2. Cold Features]
Add a field for subheading in article
Until now, you could only enter one heading for each article. Today, however, it is common to have an additional subheading that appears above the main heading.
This could be realised using custom fields or HTML. However, this is too cumbersome and / or too difficult for editors who simply have to write articles.

**Comment by chmst** [comment]
For my understanding: What is cumbersome with a custom field? 

In any case, the user has to edit the title and a subtitle in different input fields. 

If this input field is a custom field you could make an override of the article edit view in atum and your template, so that the subtitle field is placed in the edit form. 

The subtitle itself will be always part of the <h1 or <h2 other wise it could be an accessibility issue. The subtitle is only differently styled, like 
```
<h1>
<span class="subtitle">subtitle</span>
<span class="h1">title</span>
</h1>

```

**Reply by ebbes-isch-emmer** [reply]
The procedure you have described is possible. You can do a lot with custom fields.

If Joomla! is supposed to be an editing tool, it is not the point of the exercise for the admin to create overrides first.
Subheadings are standard in articles today. Why not implement this as a basic function, i.e. in the core? After all, it is not an exotic function.

**Reply by brianteeman** [reply]
I would never make a subtitle like that. Good heirarchical content is essential for seo, accessibility, marketing etc etc.
```
<h1>Heading</h1>
<h2>Sub-heading or tagline</h2>
<p>body content</p>
```

The part about custom fields not being "friendly" is that they are on a different tab in the editor view. This places them outside of the regular flow for creating content on a site that is using them.

The only solution for that is to create your own article edit view. This is easy to do and documented very well in many places online including the Joomla Magazine.

As for adding it as a core field - it might be a cmmon requirment for you but probably not for others. The UI can only ever be a compromise of providing what most people need as a minimum and the ability to add additonal functionality if needed.

**Comment by chmst** [comment]
> I would never make a subtitle like that. Good heirarchical content is essential for seo, accessibility, marketing etc etc.

 ```
<h1>Heading</h1>
<h2>Sub-heading or tagline</h2>
<p>body content</p>

```
I know from the german forum that @ebbes-isch-emmer want to place the subtitle ABOVE the main title. And for that it would be definitely wrong to place <h2 above <h1

**Reply by ebbes-isch-emmer** [reply]
Since Joomla! uses Bootstrap, you can do the following to "not harm" SEO:

```
<h1 class="subheading">...</h1>
<h2>...</h2>
```

`<h1 class="subheading">` is then smaller than h2. (defined in CSS)

**Reply by chmst** [reply]
You also can use js or a plugin - but your question was orginally "Should Joomla add subtitles to articles in the core"? 
My opinion is "no" because You never will be able to satisfy everyone and we have custom fields for inidividual solutions. Let's wait for other opinions
